var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -168,19 +189,20 @@ def cmake_args(self):
         blas = spec['blas'].libs
         lapack = spec['lapack'].libs
         options.extend([
-            '-DTrilinos_ENABLE_ALL_PACKAGES:BOOL=ON',
-            '-DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON',
             '-DTrilinos_VERBOSE_CONFIGURE:BOOL=OFF',
             '-DTrilinos_ENABLE_TESTS:BOOL=OFF',
             '-DTrilinos_ENABLE_EXAMPLES:BOOL=OFF',
             '-DCMAKE_BUILD_TYPE:STRING=%s' % (
                 'DEBUG' if '+debug' in spec else 'RELEASE'),
             '-DBUILD_SHARED_LIBS:BOOL=%s' % (
                 'ON' if '+shared' in spec else 'OFF'),
-            '-DTPL_FIND_SHARED_LIBS:BOOL=%s' % (
-                'ON' if '+shared' in spec else 'OFF'),
-            '-DTrilinos_LINK_SEARCH_START_STATIC:BOOL=%s' % (
-                'OFF' if '+shared' in spec else 'ON'),
+            # The following can cause problems on systems that don't have
+            # static libraries available for things like pthreads for example
+            # even if you want to built trilinos as static
+            # '-DTPL_FIND_SHARED_LIBS:BOOL=%s' % (
+            #     'ON' if '+shared' in spec else 'OFF'),
```

Comment:
not sure about that one, can you expand on the problem you see? You are building `~shared` trilinos and something does not work? Then i would say you can keep 
```
-DTPL_FIND_SHARED_LIBS:BOOL=
```
and only comment out 
```
-DTrilinos_LINK_SEARCH_START_STATIC
```